space-cli/cmd/upgrade.go

```
@@ -0,0 +1,246 @@
+		if err != nil {
+			logrus.Errorf("error deleting service in docker unable to list containers - %s", err)
+			return err
+		}
+		if len(containers) != 0 {
+			logrus.Errorf("Container (%s) already exists", c.containerName)
+			return fmt.Errorf("container (%s) already exists", c.containerName)
+		}
+
+		resp, err := cli.ContainerCreate(ctx, &container.Config{
+			Image:        c.containerImage,
+			ExposedPorts: c.exposedPorts,
+			Env:          c.envs,
+		}, &container.HostConfig{
+			Mounts:       c.mount,
+			PortBindings: c.portMapping,
+		}, nil, c.containerName)
+		if err != nil {
+			logrus.Errorf("Unable to create container (%s) - %s", c.containerName, err)
+			return err
+		}
+
+		if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+			logrus.Errorf("Unable to start container (%s) - %s", c.containerName, err.Error())
+			return err
+		}
+
+		// get the ip address assigned to container
+		data, err := cli.ContainerInspect(ctx, c.containerName)
+		if err != nil {
+			logrus.Errorf("Unable to inspect container (%s) - %s", c.containerName, err)
+		}
+		// Remove the domain from the hosts file
+		hosts.RemoveHost(c.dnsName)
+		// Add it back with the new ip address
+		ip := data.NetworkSettings.Networks["bridge"].IPAddress
+
+		hosts.AddHost(ip, c.dnsName)
+	}
+
+	if err := hosts.Save(); err != nil {
+		logrus.Errorf("Unable to save host file - %s", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func getLatestVersion(version string) (string, error) {
+	mongoConn := scClient.New("test", "localhost:4122", false).DB("mongo")
+	if mongoConn == nil {
+		return "", fmt.Errorf("cannot connect to mongo")
+	}
+	ctx := context.Background()
+
+	var result *spaceApiTypes.Response
+	if version == "" {
+		result, _ = mongoConn.Get("table").Select(map[string]int32{"version_no": 2}).Apply(ctx)
+	} else {
+		result, _ = mongoConn.Get("table").Where(spaceApiTypes.Cond("compatible_version", "==", version)).Apply(ctx)
```

Don't ignore the error